sunpy/io/tests/test_fits.py

```diff
@@ -176,3 +176,11 @@ def test_warn_longkey():
                                'goodkey': 'test'})
     assert 'GOODKEY' in fits.keys()
     assert 'BADLONGKEY' not in fits.keys()
+
+
+def test_read_memmap():
+    # Test that memmap is passed correctly to the FITS reader
+    data, header = sunpy.io._fits.read(AIA_171_IMAGE, memmap=True)[0]
+    # Create memmap array
+    mm = np.memmap(AIA_171_IMAGE, dtype='float32', mode='r', shape=data.shape)
```

Comment:
The test should verify that passing memmap=True to the read function opens the file as a memmap. By calling np.memmap explicitly, I don't think this actually tests the memmap functionally of the read function. Was the "data" variable not a subclass of some memmap class?

Reply:
data = <class 'numpy.ndarray'>
header = <class 'sunpy.io.header.FileHeader'>